### PR TITLE
Local development and testing instructions in the development guide

### PR DESCRIPTION
**Pull request by szczepiq**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -10,6 +10,29 @@ They are not set in stone.
...
+ - Release notes content: ```./gradlew previewReleaseNotes```
+4. Advanced testing (occasionally, for core developers, edge cases):
+ - Release notes in file: ```./gradlew previewReleaseNotes```, then inspect updated file
+ - Test release needed task: ```./gradlew assertReleaseNeeded```
+ - If you are one of the core developers you can export env variables and even test git push and bintray upload.
+ Run ```./gradlew testRelease``` follow the prompts and export necessary env variables.
+ - Test ciReleasePrepare task but beware that it will reconfigure your working copy!
+ For example, it will switch git user.
+ Remember to switch it back when you finished.
+ - Clone "mockito/mockito" repository and make testing with a proper project :)
```
Comment (mstachniuk) [inline CONTRIBUTING.md]: Please add info about `gitCommitCleanUp`, `gitTagCleanUp` and `releaseCleanUp` tasks, when run this tasks and what they do

Comment (szczepiq) [inline CONTRIBUTING.md]: Good point! I need to document what to expect and how to clean up the local working copy when things go wrong. Normally, I don't run clean up tasks, they are automatically run during 'testRelease'.

BTW. at the moment all tasks are documented in ```./gradlew tasks``` and I don't intend to include information what those tasks do in the readme. In the readme/contributing guide I plan to keep use case - based information, e.g. when to use the tasks. I will actually also document ```./gradlew tasks```.

Thank you for review!
